x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/components/enablement_modal.tsx

```diff
@@ -131,13 +147,20 @@ export const EntityStoreEnablementModal: React.FC<EntityStoreEnablementModalProp
       </EuiModalBody>
 
       <EuiModalFooter>
-        <EuiButtonEmpty onClick={() => toggle(false)}>{'Cancel'}</EuiButtonEmpty>
-        <EuiButton onClick={enableStore(enablements)} fill>
-          <FormattedMessage
-            id="xpack.securitySolution.entityAnalytics.enablements.modal.enable"
-            defaultMessage="Enable"
-          />
-        </EuiButton>
+        <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+          <EuiFlexItem>{!enablementOptions && proceedWarning}</EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup direction="row">
+              <EuiButtonEmpty onClick={() => toggle(false)}>{'Cancel'}</EuiButtonEmpty>
```

Comment:
Unrelated: This component requires some refactoring:
* use callback for the click handle.
* 'cancel' i18n
* Calling `enableStore(enablements)` during the rendering of the component is super weird. Should it be a callback?

Reply:
**enableStore** - would we prefer the callback due to that function being passed as props and it being effected by any re-render of the parent component? e.g. want to minimise unnecessary re-renders in child component (current component) - changed this to use the callback for this reason.

**toggle(false)** - it seems quite simple so it seems ok to use the anonymous function here?

**cancel** - i18n, agreed, will change this up.

Reply:
Or if I misunderstood when you say callback do you mean the hook useCallback or simply giving it a callback where I could instead use an anon function like `<EuiButton onClick={() => enableStore(enablements)} fill> ` or give it to a function outside the render such as
```
const handleEnableClick = () => {
    enableStore(enablements);  }; 
```
 and call like `<EuiButton onClick={handleEnableClick} fill>`

Reply:
I am okay with the other items. But `enableStore` is called during every render. Every time the component is rendered, it is called the function.

For some reason, it is declared as a function that return another function: 
`const enableEntityStore = (enable: Enablements) => () => {`
Why? This is an uncommon react pattern for a callback.

Reply:
I approved the PR since this change is unrelated.